### PR TITLE
Updates docs to show the 'Result' type

### LINKED ISSUE #385: Readme/Docs are broken
They refer to old callback parameters that don't exist. See: https://github.com/Moya/Moya/issues/384

### PR DESCRIPTION
**Pull request by colinta**
Hopefully this resolves #385. @yinanfang Does this clarify things?

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -80,9 +80,17 @@ Use
 After [some setup](docs/Examples.md), using Moya is really simple. You can access an API like this:
 
 ```swift
-provider.request(.Zen) { (data, statusCode, response, error) in
-    if let data = data {
-        // do something with the data
+provider.request(.Zen) { result in
+    switch result {
+    case let .Success(moyaResult):
```
Comment (colinta) [inline Readme.md]: `result` and `moyaResult`?  Confusing variable names?  This is what I called these in the Ello app, but I'm open to suggestions for better names!

File: Readme.md [hunk 2]
```diff
@@ -80,9 +80,17 @@ Use
...
+        let data = moyaResult.data
+        let statusCode = moyaResult.statusCode
+        // do something with the response data or statusCode
+    case .Failure(error):
+        // this means there was a network failure - either the request
+        // wasn't sent (connectivity), or no response was received (server
+        // timed out).  If the server responds with a 4xx or 5xx error, that
+        // will be sent as a ".Success"-ful response.
```
Comment (colinta) [inline Readme.md]: Trying to be helpful here, in light of #379.

**Comment by ashfurrow** [comment]
Cool, thanks! I think we need to apply this same fix in https://github.com/Moya/Moya/blob/master/docs/Providers.md and https://github.com/Moya/Moya/blob/master/docs/Examples.md too.

**Comment by colinta** [comment]
Ahh gotcha.  Will do.

**Comment by ashfurrow** [comment]
:bow:

**Comment by colinta** [comment]
@ashfurrow phew, went a little nuts on `Examples.md`: ab85328

**Comment by colinta** [comment]
https://github.com/Moya/Moya/blob/ab853283d8463e53fe306ed2b52b623f9db88856/docs/Examples.md

**Comment by ashfurrow** [comment]
Looks amazing! 😻

**Comment by yinanfang** [comment]
@colinta @ashfurrow Thank you so much for the update! I just came across another problem. I got `kCFStreamErrorDomainSSL, -9843` error. I found that this happened because the Apple ATS which doesn't allows insecure HTTPS call. Alamofire suggested an solution here: https://github.com/Alamofire/Alamofire#app-transport-security & https://github.com/Alamofire/Alamofire/issues/977#issuecomment-167925391 
Do you know how I can configure the network manager in Moya similarly? Should I open another issue for this?

**Comment by yinanfang** [comment]
Oops... I just found the documentation here: https://github.com/Moya/Moya/blob/master/docs/Providers.md#advanced-use Sorry about that. It's not a very smart question.

**Pull request merged**